File: Python/Test/jdi_uitest_web_examples/main/site/epam/epam_site.py
from JDI.core.interfaces.check_page_types import CheckPageTypes
from JDI.web.selenium.elements.api_interact.find_element_by import By
from JDI.web.selenium.elements.common.button import Button
from JDI.web.selenium.elements.complex.menu import Menu
from JDI.web.selenium.elements.composite.web_site import WebSite
from Test.jdi_uitest_web_examples.main.enums.enums import HeaderMenu, HeaderSolutionsMenu
from Test.jdi_uitest_web_examples.main.site.epam.pages import HomePage, CareerPage, ProductDevelopmentPage, \
    JobListingPage, JobDescriptionPage


class EpamSite(WebSite):
    home_page = HomePage(url="/", title="EPAM | Software Product Development Services")
    career_page = CareerPage(url="/careers", title="Careers")
    product_development_page = ProductDevelopmentPage(url="/solutions/core-engineering/product-development")
    job_listing_page = JobListingPage(
         url="/careers/job-listings?sort=best_match&query=Engineer&department=Software+Test+Engineering&city=St-Petersburg&country=Russia",
         url_template="/careers/job-listings", title="Job Listings",
         url_check_type=CheckPageTypes.CONTAINS, title_check_type=CheckPageTypes.CONTAINS)
    job_description_page = None #JobDescriptionPage(
        # url="/careers/job-listings/job.11584#apply", urlTemplate=".*/careers/job-listings/job\\.\\d*#apply",
        # urlCheckType=CheckPageTypes.MATCH)

    header_menu = Menu(by_all_options_names_locator=By.css(".tile-menu>li>a"), parametrized_class=HeaderMenu)
    multiple_header_menu = Menu(by_menu_levels_locators=[By.css(".tile-menu>li>a"),
                                                         By.xpath("//*[@class='tile-menu']//*[@href='/solutions']//..//li")])

    line_menu = None#JDIElements(By.css(".tile-menu>li>a"), Button)

    header_solutions_menu = None #Menu(By.css(".tile-menu .submenu a"), HeaderSolutionsMenu)

File: Java/Tests/jdi-uitest-tutorialtests/src/main/java/com/epam/jdi/site/epam/EpamSite.java
package com.epam.jdi.site.epam;

import com.epam.jdi.enums.HeaderMenu;
import com.epam.jdi.enums.HeaderSolutionsMenu;
import com.epam.jdi.site.epam.pages.*;
import com.epam.jdi.site.epam.sections.Header;
import com.epam.jdi.uitests.web.selenium.elements.complex.Menu;
import com.epam.jdi.uitests.web.selenium.elements.composite.WebSite;
import com.epam.jdi.uitests.web.selenium.elements.pageobjects.annotations.JFindBy;
import com.epam.jdi.uitests.web.selenium.elements.pageobjects.annotations.JPage;
import com.epam.jdi.uitests.web.selenium.elements.pageobjects.annotations.JSite;
import com.epam.jdi.uitests.web.selenium.elements.pageobjects.annotations.objects.JMenu;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.FindBy;

import static com.epam.jdi.uitests.core.interfaces.complex.tables.interfaces.CheckPageTypes.CONTAINS;
import static com.epam.jdi.uitests.core.interfaces.complex.tables.interfaces.CheckPageTypes.MATCH;

/**
 * Created by Roman_Iovlev on 8/30/2015.
 */
@JSite("https://www.epam.com")
public class EpamSite extends WebSite {

    @JPage(url = "/", title = "EPAM|Software Product Development Services")
    public static HomePage homePage;

    @JPage("/careers")
    public static CareerPage careerPage;
    public static ProductDevelopmentPage productDevelopmentPage;

    @JPage(url = "/careers/job-listings?query=test&country=Russia&city=St-Petersburg&department=Software+Test+Engineering",
            urlTemplate = "/careers/job-listings", title = "Job Listings",
            urlCheckType = CONTAINS, titleCheckType = CONTAINS)
    public static JobListingPage jobListingPage;


    @JPage(url = "/careers/job-listings/job.11516", urlTemplate = ".*/careers/job-listings/job\\.\\d*",
            urlCheckType = MATCH)
    public static JobDescriptionPage jobDescriptionPage;

    @JPage(url = "/our-work", title = "Our Work")
    public static ourWorkPage ourWorkPage;

    @JPage(url = "/contact-us", title = "Contact Us | EPAM Systems")
    public static ContactUs contactUs;

    @JPage(url = "/insights", title = "Insights")
    public static InsightsPage insightsPage;

    @FindBy(css = ".top-navigation__list")
    public static Menu<HeaderMenu> headerMenu;

    @JMenu(level1 = @JFindBy(css = "ul.top-navigation__list>li span a"),
           level2 = @JFindBy(css = "ul.top-navigation__grand-sub-list>li a"))
    public static Menu multipleHeaderMenu;

    @FindBy(css = ".tile-menu .submenu a")
    public static Menu<HeaderSolutionsMenu> headerSolutionsMenu = new Menu<HeaderSolutionsMenu>() {

        @Override
        protected void selectAction(String name) {
            Actions action = new Actions(getDriver());
            WebElement el = getDriver().findElements(By.cssSelector(".tile-menu a")).get(0);
            action.moveToElement(el).build().perform();
            super.selectAction(name);
        }
    };

    public static Header header;
}
